symengine/dict.h

```diff
@@ -43,6 +43,18 @@ typedef std::map<RCP<const Integer>, unsigned, RCPIntegerKeyLess>
     map_integer_uint;
 typedef std::map<unsigned, integer_class> map_uint_mpz;
 typedef std::map<int, Expression> map_int_Expr;
+//! Part of umap_vec_mpz:
+typedef struct {
+    inline std::size_t operator()(const vec_int &k) const
+    {
+        std::size_t h = 0;
+        for (const auto &p : k) {
+            h = (h << 4) + p;
```

Comment:
This was being used before, I've just moved it to the top, near the typedefs. Should I change it?